ister.py

```diff
@@ -408,43 +411,62 @@ def get_current_format():
+    LOG.info("Starting {0}. May take several minutes".format(package_manager))
+
+    if package_manager == SWUPD:
+        if not args.format:
+            args.format = get_current_format()
+        if args.fast_install:
+            args.statedir = "{0}/tmp/swupd".format(target_dir)
+        add_bundles(template, target_dir)
+        cmd = "swupd verify --install"
+        cmd += " --path={0}".format(target_dir)
+        cmd += " --manifest={0}".format(template["Version"])
+        cmd += " --contenturl={0}".format(args.contenturl)
+        cmd += " --versionurl={0}".format(args.versionurl)
+        cmd += " --format={0}".format(args.format)
+        cmd += " --statedir={0}".format(args.statedir)
+        if args.cert_file:
+            cmd += " -C {0}".format(args.cert_file)
+    else:
+        if not os.path.exists(args.dnf_config):
+            add_default_dnf_config(args)
+        cmd = "dnf install --assumeyes"
+        cmd += " --config {0}".format(args.dnf_config)
+        cmd += " --installroot {0}".format(target_dir)
+        cmd += " {0}".format(" ".join(template["Bundles"]))
     if shutil.which("stdbuf"):
-        swupd_command = "stdbuf -o 0 {0}".format(swupd_command)
-    swupd_command += " --contenturl={0}".format(args.contenturl)
-    swupd_command += " --versionurl={0}".format(args.versionurl)
-    swupd_command += " --format={0}".format(args.format)
-    if args.fast_install:
-        args.statedir = "{0}/tmp/swupd".format(target_dir)
-    swupd_command += " --statedir={0}".format(args.statedir)
-    if args.cert_file:
-        swupd_command += " -C {0}".format(args.cert_file)
-
-    if template["DestinationType"] == "physical":
+        cmd = "stdbuf -o 0 {0}".format(cmd)
+
+    if package_manager == SWUPD and template["DestinationType"] == "physical":
         os.makedirs(args.statedir, exist_ok=True)
         os.chmod(args.statedir, stat.S_IRWXU)
         os.makedirs("{0}/var/tmp".format(target_dir))
         os.chmod("{0}/var/tmp".format(target_dir), stat.S_IRWXU)
-        run_command("mount --bind {0}/var/tmp {1}"
-                    .format(target_dir, args.statedir))
-    swupd_env = os.environ
+        run_command("mount --bind {0}/var/tmp {1}".format(target_dir, args.statedir))
+
+    cmd_env = os.environ
     if template.get("HTTPSProxy"):
-        swupd_env["https_proxy"] = template["HTTPSProxy"]
+        cmd_env["https_proxy"] = template["HTTPSProxy"]
         LOG.debug("https_proxy: {}".format(template["HTTPSProxy"]))
+    
+    run_command(cmd, environ=cmd_env, show_output=True)
 
-    run_command(swupd_command, environ=swupd_env, show_output=True)
-
-    if args.fast_install:
+    if package_manager == SWUPD and args.fast_install:
         run_command("rm -rf {0}".format(args.statedir))
+    if package_manager == DNF and os.path.exists("{0}/usr/bin/update-helper".format(target_dir)):
+        run_command("{0}/usr/bin/update-helper {1}".format(target_dir, target_dir))
```

Comment:
This should be part of the post-dnf script.

Reply:
Ah, yes.  I'll move this in.